### DISCUSSION TITLE
How to extract trace data from jaeger

### DISCUSSION THREAD

**Discussion by gxxxh** [jaegertracing/jaeger#2876] [Q&A] [answered]
I am trying to learn kubernetes recently. I have already deployed jaeger (all-in-one) on istio. Although I can see the trace information on the jaeger UI, I don't know how to extract these trace data by python so that I can use these data to do something about AIops. 
I think there should be some API to access these data dicectly by python but I don't find it. or can I access the cassandra using python to get these data. I don't find how to do it so I open this discussion.

**Comment by jpkrohling** [comment]
@Ashmita152 might be able to help you on this.

**Comment by yurishkuro** [comment]
The APIs are documented here: https://www.jaegertracing.io/docs/1.22/apis/

**Comment by Ashmita152** [comment, marked as answer]
Hi @gxxxh 

There are various ways to pull jaeger data as Yuri shared. The easiest one I found is by hitting jaeger-query's http endpoint: /api/traces. You can refer to this python script to do the same: 

https://github.com/Ashmita152/jaeger-datasets/blob/master/bookinfo/extract.py

Let us know if you have more questions. Thank you.

**Reply by gxxxh** [reply]
Thank you very much for the answer. I have realized that I had got into a completely wrong direction. The answer is really helpful.

**Reply by kapilagrawal95** [reply]
Hi, this does not seem to work now. Get the following error when running your script for my use-case:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

**Reply by Entromorgan** [reply]
@kapilagrawal95 I have the same error. Changing the endpoints from "http://localhost:16686/jaeger/api/xxx" to "http://localhost:16686/api/xxx" can solve it.
